### PR TITLE
[kube-prometheus-stack] Add exemplars related settings

### PR DESCRIPTION
**Pull request by faustodavid**
Signed-off-by: Fausto David Suarez Rosario <faustodavid@hotmail.com>

#### What this PR does / why we need it
Add exemplars related settings. This PR enables the following feature released on **prometheus-operator** version **0.58.0**.
https://github.com/prometheus-operator/prometheus-operator/pull/4834

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 39.2.1
+version: 39.2.2
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]: new feature should raise minor version

File: charts/kube-prometheus-stack/values.yaml [hunk 2]
```diff
@@ -2286,6 +2286,14 @@ prometheus:
...
     ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#webtlsconfig
     web: {}
+    
+    ## Exemplars related settings that are runtime reloadable.
+    ## It requires to enable the exemplar storage feature to be effective.
+    exemplars:
+      # Maximum number of exemplars stored in memory for all series.
+      # If not set, Prometheus uses its default value.
+      # A value of zero or less than zero disables the storage.
+      maxSize: 100000
```
Comment (monotek) [inline charts/kube-prometheus-stack/values.yaml:2296]: why not using the default?

```suggestion
      maxSize: ""
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/values.yaml [hunk 3]
```diff
@@ -2286,6 +2286,14 @@ prometheus:
     ## WebTLSConfig defines the TLS parameters for HTTPS
     ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#webtlsconfig
     web: {}
+    
+    ## Exemplars related settings that are runtime reloadable.
+    ## It requires to enable the exemplar storage feature to be effective.
+    exemplars:
```
Comment (monotek) [inline charts/kube-prometheus-stack/values.yaml:2292]:
```suggestion
    exemplars: ""
```

File: charts/kube-prometheus-stack/values.yaml (hunk 2 above)
Comment (monotek) [inline charts/kube-prometheus-stack/values.yaml:2296]:
```suggestion
      # maxSize: 100000
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml [hunk 4]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 39.2.1
+version: 39.3.1
```
[the remark this answers is not in the document]
Comment (faustodavid) [reply charts/kube-prometheus-stack/Chart.yaml]: why from 39.2.1 to 39.4.0, shouldn't it be 39.3.0?

**Review by monotek — APPROVED** [review]

**Pull request merged**